src/components/flowchart/flowchart.js

```diff
@@ -134,6 +135,15 @@ export class FlowChart extends Component {
         this.updateSlicedPipelineState(from, to, this.props.slicedPipeline);
       }
     }
+
+    // Hide slicing notification if metadata panel is closed using button
```

Comment:
I did not fully understand the condition here. Can you please add comments or let me know what these conditions are for. Thank you

Reply:
The first condition is checking if a previous node was clicked, `prevProps.clickedNode`, so the metadata panel was opened and it checks if a current node has not been clicked, `!this.props.clickedNode`,  which implies we have closed the metadata panel. This is the scenario where we have shift+clicked a single node which results in metadata-panel opening and user closing the metadata panel using the X button.

The second condition, `!this.props.slicedPipeline.length`, just makes sure we run this during slicing.

This scenario is where we have shift+clicked a single node which results in metadata-panel opening and user closing the metadata panel using the X button. We want to reset the slicing because without this it was keeping the node highlighted.

Reply:
@Huongg the bug where the node remains highlighted until we hover over a different node after metadata panel is closed is still present.

Reply:
hey @SajidAlamQB shouldn't it just to set the notification to false, we don't need to set the slicingPipelineState as well?

```
// Hide slicing notification when metadata panel is closed
if (this.props.clickedNode !== prevProps.clickedNode && !this.props.clickedNode) {
  this.setState({ showSlicingNotification: false });
}
```

Reply:
and yes that bug is fixed in the other PR, you can check the demo branch to see if you can still reproduce the bug

Reply:
Ah yes it works on the demo branch 💯 and we don't need `slicingPipelineState` it was left over from when I was trying to fix the bug.

Reply:
> This scenario is where we have shift+clicked a single node which results in metadata-panel opening

Is this by design ? or can we do something like to enter the slice mode, user need to click on nodes using shift + click. Why should we open the metadata-panel when someone is using shift+click on a node ? Possibly they might want to use the slice functionality.

I would suggest -

1. In regular mode - 
   a. If the user clicks on a node without using shift, we open the metadata panel. 
   b. If the user clicks on a node using shift, even if it is a one node click, we do not open the metadata panel and wait for a second node click using shift
2. In slice mode view -
   a. If the user clicks on a node with/without using shift, we open the metadata panel (as we do not allow multi-slicing at this moment)
   
   In summary, I feel we should not open a metadata panel when someone is shift+clicking on a node.
   
   Thank you

Reply:
After looking at this PR - https://github.com/kedro-org/kedro-viz/pull/1980
The scenario I suggested above already exists (if I click on first node using shift, it does not open metadata panel). 

What do you mean by - `This scenario is where we have shift+clicked a single node which results in metadata-panel opening` @SajidAlamQB in your response above ? Thank you

Reply:
okay, I understand the condition now. But how should the notification behave in the below use case (in gif) -

1. I click on a node -> the notification appears + metadata panel opens
2. I shift+click on another node -> the notification now says number of nodes selected + metadata panel closes
3. Now I click on a different node -> At this moment, this opens the metadata panel but does not show the notification as it shows in step 1 (which seems inconsistent). If the behavior is consistent, this is just re-iteration of step 1, so it should do -> the notification appears + metadata panel opens

**For design -**
In the gif, if you see the initial notification appears at a different position and then after selection of 2 nodes, the notification position changes. I feel it would be better if we just change the text content of the already present notification bar and do not re-run the animation. Wdyt @stephkaiser 

![shift_click_scenario_notification](https://github.com/user-attachments/assets/932546f5-2b1d-47f6-ba09-89550f50e184)

Happy to discuss on this further. Thank you

Reply:
> What do you mean by - `This scenario is where we have shift+clicked a single node which results in metadata-panel opening` @SajidAlamQB in your response above ? Thank you

Sorry I meant just clicked node, not shift+click. When you click a node the metadata panel opened and if user closed it with 'X' the notification stayed. That's why the condition is there.

Reply:
> 1. I click on a node -> the notification appears + metadata panel opens
> 2. I shift+click on another node -> the notification now says number of nodes selected + metadata panel closes
> 3. Now I click on a different node -> At this moment, this opens the metadata panel but does not show the notification as it shows in step 1 (which seems inconsistent). If the behavior is consistent, this is just re-iteration of step 1, so it should do -> the notification appears + metadata panel opens

This is fixed please test it and lmk your thoughts.